numba/cpython/charseq.py

```diff
@@ -204,9 +204,12 @@ def _make_constant_bytes(context, builder, nbytes):
     bstr.itemsize = ir.Constant(bstr.itemsize.type, 1)
     bstr.data = context.nrt.meminfo_data(builder, bstr.meminfo)
     bstr.parent = cgutils.get_null_value(bstr.parent.type)
-    # bstr.shape and bstr.strides are not used
-    bstr.shape = cgutils.get_null_value(bstr.shape.type)
-    bstr.strides = cgutils.get_null_value(bstr.strides.type)
+    bstr.shape = cgutils.pack_array(builder, [bstr.nitems],
+                                    context.get_value_type(types.intp))
```

Comment:
I think that this results in a pessimistic description of the "shape" as the always positive nature of the values is not being encoded. However, given that the bytes types shouldn't really be inheriting from arrays and work is on going to correct that, it probably isn't worth fixing.